src/goto-symex/field_sensitivity.h

```diff
@@ -15,7 +15,36 @@ class namespacet;
 class goto_symex_statet;
 class symex_targett;
 
-/// \brief Control granularity of object accesses
+/// Control granularity of object accesses
+///
+/// [Field sensitivity](\ref field_sensitivityt) is a transformation of the
+/// instructions goto-program which mainly replaces some expressions by symbols
+/// but can also add assignments to the target equations produced by symbolic
+/// execution.
```

Comment:
This is quite a vague description, suggest mentioning that it replaces member operators with atomic symbols representing a field when possible, and when it cannot it adds extra assignments to convert whole-struct operations into several writes to individual fields.

Reply:
+1 - I think we should separate the high-level aim (during member/index expressions into atomic symbols) from implementation details (fixing up in the failing case). That kind of "separation" might just be done by adding a paragraph in between.

Reply:
@tautschnig @smowton I added a paragraph with a high level description before getting into the details, can you check it? (last commit)